numhask/src/NumHask/Algebra/Abstract/Additive.hs

```diff
@@ -1,78 +1,58 @@
-{-# OPTIONS_GHC -Wall #-}
-{-# LANGUAGE ConstrainedClassMethods #-}
-{-# LANGUAGE ConstraintKinds #-}
 {-# LANGUAGE DeriveFunctor #-}
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE GADTs #-}
 {-# LANGUAGE MonoLocalBinds #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE RankNTypes #-}
-{-# LANGUAGE RoleAnnotations #-}
-{-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wall #-}
 
--- | The Group hirarchy
+-- | Additive
 module NumHask.Algebra.Abstract.Additive
-      ( zero
-      , Sum(..)
-      , coerceFA
-      , coerceFA'
-      , coerceTA
-      , coerceTA'
-      , Addition(..)
-      , (+)
-      , (-)
-      , negate
-      , plus
-      )
+  ( Sum(..)
+  , Additive
+  , (+)
+  , (-)
+  , zero
+  , negate
+  , sum
+  )
 where
 
-import qualified Prelude                       as P
-import qualified GHC.Generics                  as P
-import           NumHask.Algebra.Abstract.Group
 import Data.Coerce
-import           Data.Complex                   ( Complex(..) )
-import           Data.Int                       ( Int8
-                                                , Int16
-                                                , Int32
-                                                , Int64
-                                                )
-import           Data.Word                      ( Word
-                                                , Word8
-                                                , Word16
-                                                , Word32
-                                                , Word64
-                                                )
-import           GHC.Natural                    ( Natural(..) )
+import Data.Complex (Complex(..))
+import Data.Int (Int8, Int16, Int32, Int64)
+import Data.Word (Word, Word8, Word16, Word32, Word64)
+import GHC.Natural (Natural(..))
+import NumHask.Algebra.Abstract.Group
+import qualified GHC.Generics as P
+import qualified Prelude as P
 
 newtype Sum a = Sum a
-      deriving (P.Eq, P.Ord, P.Read, P.Show, P.Bounded, P.Generic, P.Generic1, P.Functor)
-
-class (Associative (Sum a), Commutative (Sum a), Unital (Sum a)) => Addition a where
-      sum :: (P.Foldable f) => f a -> a
-      sum = P.foldr (+) zero
-instance (Associative (Sum a), Commutative (Sum a), Unital (Sum a)) => Addition a
+  deriving (P.Eq, P.Ord, P.Read, P.Show, P.Bounded, P.Generic, P.Generic1,
+            P.Functor)
 
-plus :: Magma (Sum a) => a -> a -> a
-plus = coerceFA magma
+class (Associative (Sum a), Commutative (Sum a), Unital (Sum a)) => Additive a
```

Comment:
You moved `sum ` out of Addition. I know `sum`, as it was formulated above, was not the best idea, but I think we need to solve it. Making special instances for `sum` and `product` is on my list, they are very important for stable numeric calculations.

We can decide to delete them, because the current solution is so bad, but then we need to reopen #70

Reply:
#100 reverts